### PR TITLE
Test/invite users

### PR DESCRIPTION
**Pull request by ramirobg94**
## Related Issue

Resolves 1395 / invite user 

## Brief description of the changes proposed an/or how the issue(s) has(have) been solved.

add test to invite user

## How has this been tested

this is a test

##### Do you consider this PR needs further testing?
- [x] No
- [ ] Yes

## Types of changes

- [ ] Docs change 
- [ ] Refactoring 
- [ ] Dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Disclaimer

Captura de pantalla 2021-02-25 a las 23 28 07

### CODE REVIEW

File: webapp/utils/dataTestId/index.js [hunk 1]
```diff
@@ -118,4 +118,11 @@ export const DataTestId = {
     taxonomyDescription: _withLang('taxonomy-description'),
     uploadInput: 'taxonomy-upload-input',
   },
+  userList: {
+    inviteBtn: 'invite-user-btn',
+    email: 'invite-user-name',
+    group: 'invite-user-group',
+    groupByName: (name) => `div[text="${name}"]`,
```
Comment (minotogna) [inline webapp/utils/dataTestId/index.js:125]: this should return just the id. then in the test you can use ```getSelector(DataTestId.(groupByName))```.

File: webapp/app/appModules.js [hunk 2]
```diff
@@ -121,7 +121,7 @@ export const dataModules = {
 export const userModules = {
   users: {
     key: 'userList',
-    path: `${appModules.users.path}/list`,
+    path: `${appModules.users.path}/userList`,
```
Comment (minotogna) [inline webapp/app/appModules.js:124]: any reason to change the path?

File: test/e2e_playwright/tests/surveyInviteUser.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  test(`Invite user`, async () => {
+    await page.click(getSelector(DataTestId.userList.inviteBtn, 'a'))
+
+    await page.fill(getSelector(DataTestId.userList.email, 'input'), 'testtwo@arena.com')
+
+    await page.click(getSelector(DataTestId.dropdown.toggleBtn(DataTestId.userList.group), 'button'))
+    await page.click(`text="Survey administrators"`)
+
+    await page.click(getSelector(DataTestId.userList.submitBtn, 'button'))
+    await page.waitForTimeout(2500)
```
Comment (minotogna) [inline test/e2e_playwright/tests/surveyInviteUser.js:25]: better to wait for response like:

```
await Promise.all([ 
    page.waitForResponse('**/users'), // check the right path though 
    page.click(getSelector(DataTestId.userList.submitBtn, 'button'))
] )
```

**page.waitForTimeout** should be used mainly for debugging https://playwright.dev/docs/api/class-page#pagewaitfortimeouttimeout

File: test/e2e_playwright/tests/surveyInviteUser.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+export default () =>
+  describe('Invite user testtwo@arena.com', () => {
+    gotoSurveyUsersList()
+    expectUsers({ users: ['test@arena.com'] })
+
+    inviteUser()
+
+    expectUsers({ users: ['test@arena.com', 'testtwo@arena.com'] })
+  })
```
Comment (minotogna) [inline test/e2e_playwright/tests/surveyInviteUser.js:37]: last step should be ```gotoHome()```

File: test/e2e_playwright/tests/surveyInviteUser.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import { DataTestId, getSelector } from '../../../webapp/utils/dataTestId'
+import { gotoSurveyUsersList } from './_navigation'
+
+const expectUser = async ({ user }) => {
+  await expect(page).toHaveText(user)
```
Comment (minotogna) [inline test/e2e_playwright/tests/surveyInviteUser.js:5]: you can access the specific row and then check the right column
```
const emailEl = await page.$('div[data-testid="users-${0}"] div[data-testid="user-email"]')
expect(await emailEl.getInnerText()).toBe(user))
```

**Review by minotogna — CHANGES_REQUESTED** [review]

File: webapp/app/appModules.js (hunk 2 above)
Comment (ramirobg94) [reply webapp/app/appModules.js:124]: it is the only module in which the key is not the same as the submodule. is there any reason for that? 

```
export const userModules = {
  users: {
    key: 'userList',
    path: `${appModules.users.path}/list`,
  },
  user: {
    key: 'user',
    path: `${appModules.users.path}/user`,
  },
  userInvite: {
    key: 'userInvite',
    path: `${appModules.users.path}/userInvite`,
  },
}
```

File: test/e2e_playwright/tests/surveyInviteUser.js (hunk 3 above)
Comment (ramirobg94) [reply test/e2e_playwright/tests/surveyInviteUser.js]: Changed, thanks

File: test/e2e_playwright/tests/surveyInviteUser.js (hunk 4 above)
Comment (ramirobg94) [reply test/e2e_playwright/tests/surveyInviteUser.js:44]: Done

File: test/e2e_playwright/tests/surveyInviteUser.js (hunk 5 above)
Comment (ramirobg94) [reply test/e2e_playwright/tests/surveyInviteUser.js]: changed

**Review by minotogna — APPROVED** [review]

**Review by SteRiccio — APPROVED** [review]

**Pull request merged**